### PR TITLE
Improve IB error messages in GetAccountHoldings and GetOpenOrders

### LINKED ISSUE #2441: Improve IB error messages in GetAccountHoldings and GetOpenOrders
#### Expected Behavior
Errors such as invalid security types should include a more descriptive error message.

#### Actual Behavior
In some cases the error is hidden by a generic 15 second timeout error message.

#### Potential Solution
Improve error handling in both `GetAccountHoldings` and `GetOpenOrders` methods.

#### Reproducing the Problem
n/a

#### System Information
n/a

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

### PR DESCRIPTION
**Pull request by StefanoRaggi**
#### Description
- Added full contract description in `OpenOrderEventArgs.ToString`
- Improved error handling and messaging in brokerage initialization
- Updated `ConvertSecurityType` error message for unsupported security types
- Update catch block in Engine to include inner exceptions instead of current stack trace

#### Related Issue
Closes #2441 

#### Motivation and Context
Depending on the error type, occasional incomplete error messages and/or stack traces.

#### Requires Documentation Change
No.

#### How Has This Been Tested?
Local and cloud exceptions during brokerage initialization (`GetAccountHoldings` and `GetOpenOrders`).

#### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Brokerages/InteractiveBrokers/InteractiveBrokersBrokerage.cs [hunk 1]
```diff
@@ -1609,9 +1640,26 @@ private void EmitOrderFill(Order order, Execution execution, CommissionReport co
...
+            }
+            catch (Exception exception)
+            {
+                Log.Error($"InteractiveBrokersBrokerage.HandlePortfolioUpdates(): {exception}");
+
+                if (e.Position != 0)
+                {
+                    // Force a runtime error only with a nonzero position for an unsupported security type,
+                    // because after the user has manually closed the position and restarted the algorithm,
+                    // he'll have a zero position but a nonzero realized PNL, so this event handler will be called again.
```
Comment (jaredbroad) [inline Brokerages/InteractiveBrokers/InteractiveBrokersBrokerage.cs]: Nice comment and handling.

File: Brokerages/InteractiveBrokers/Client/SecurityType.cs [hunk 2]
```diff
@@ -75,6 +75,11 @@ public static class SecurityType
         /// </summary>
         public const string Bill = "BILL";
 
+        /// <summary>
+        /// Contract For Difference
+        /// </summary>
+        public const string ContractForDifference = "CFD";
```
Comment (jaredbroad) [inline Brokerages/InteractiveBrokers/Client/SecurityType.cs]: This doesn't seem used anywhere?

File: Brokerages/InteractiveBrokers/Client/OpenOrderEventArgs.cs [hunk 3]
```diff
@@ -59,7 +59,7 @@ public OpenOrderEventArgs(int orderId, Contract contract, Order order, OrderStat
         /// </summary>
         public override string ToString()
         {
-            return $"OrderId: {OrderId}, Symbol: {Contract.Symbol}, OrderStatus: {OrderState.Status}";
+            return $"OrderId: {OrderId}, Contract: {Contract}, OrderStatus: {OrderState.Status}";
```
Comment (jaredbroad) [inline Brokerages/InteractiveBrokers/Client/OpenOrderEventArgs.cs]: What is the ToString() for `Contract` class?

File: Brokerages/InteractiveBrokers/Client/SecurityType.cs (hunk 2 above)
Comment (StefanoRaggi) [reply Brokerages/InteractiveBrokers/Client/SecurityType.cs]: Yes, I just added it for completeness -- also maybe we'll add CFD support for IB in the future?

File: Brokerages/InteractiveBrokers/Client/OpenOrderEventArgs.cs (hunk 3 above)
Comment (StefanoRaggi) [reply Brokerages/InteractiveBrokers/Client/OpenOrderEventArgs.cs]:
``` C#
        public override string ToString()
        {
            return SecType + " " + Symbol + " " + Currency + " " + Exchange;
        }
```

File: Brokerages/InteractiveBrokers/InteractiveBrokersBrokerage.cs [hunk 4]
```diff
@@ -2066,57 +2114,40 @@ private static string ConvertSecurityType(SecurityType type)
...
-                case IB.SecurityType.Warrant:
-                case IB.SecurityType.Bill:
-                case IB.SecurityType.Undefined:
-                    return SecurityType.Base;
-
                 default:
-                    throw new ArgumentOutOfRangeException("type");
+                    throw new NotSupportedException(
+                        $"An existing position or open order for an unsupported security type was found: {contract}. " +
+                        "Please manually close the position or cancel the order before restarting the algorithm.");
```
Comment (mchandschuh) [inline Brokerages/InteractiveBrokers/InteractiveBrokersBrokerage.cs]: Nice,  now users will have an idea of where to start. Thank you.

File: Engine/Engine.cs [hunk 5]
```diff
@@ -184,8 +184,8 @@ public void Run(AlgorithmNodePacket job, AlgorithmManager manager, string assemb
                 catch (Exception err)
                 {
                     Log.Error(err);
-                    var runtimeMessage = "Algorithm.Initialize() Error: " + err.Message + " Stack Trace: " + err.StackTrace;
-                    _algorithmHandlers.Results.RuntimeError(runtimeMessage, err.StackTrace);
+                    var runtimeMessage = "Algorithm.Initialize() Error: " + err.Message + " Stack Trace: " + err;
+                    _algorithmHandlers.Results.RuntimeError(runtimeMessage, err.ToString());
```
Comment (mchandschuh) [inline Engine/Engine.cs]: this impacts what ends up in the database, I think we recently this is up a certain way for a reason, lemme check the history here.

**Pull request merged**